src/controllers/dynakube/oneagent/daemonset/daemonset.go

```diff
@@ -142,7 +142,10 @@ func appendHostIdArgument(result *appsv1.DaemonSet, source string) {
 func (dsInfo *builderInfo) BuildDaemonSet() (*appsv1.DaemonSet, error) {
 	instance := dsInfo.instance
 	podSpec := dsInfo.podSpec()
-	labels := kubeobjects.MergeLabels(dsInfo.buildLabels(), dsInfo.hostInjectSpec.Labels)
+	labels := kubeobjects.MergeLabels(
+		BuildLabels(dsInfo.instance.Name, dsInfo.deploymentType),
```

Comment:
I'd wrap `BuildLabels(dsInfo.instance.Name, dsInfo.deploymentType)` in `dsInfo.BuildLables()`
```suggestion
		dsInfo.BuildLabels(),
```

looks nicer